### ISSUE TITLE
Format of writing responses (nested HttpHandlers)

### ISSUE THREAD

**Issue by gerardtoconnor** [dustinmoris/Giraffe#113]
I have noticed since the adoption of the new continuation format it has made the writing of some responses a little awkward looking and very evident in recent [MS blog on Giraffe](https://blogs.msdn.microsoft.com/dotnet/2017/09/26/build-a-web-service-with-f-and-net-core-2-0/)

Issue is around needing access to ctx and not being able to do ETA reduction of child arguments of internal HttpHandlers (nested vs pipelined), eg from blog:
```fsharp
let handleLunchFilter (next: HttpFunc) (ctx: HttpContext) =
    let filter = ctx.BindQueryString<LunchFilter>()
    let lunchSpots = LunchAccess.getLunches filter
    json lunchSpots next ctx                  // Inception style HttpHandler within a HttpHandler
```
Given the fact that writing JSON / XML / TEXT to the response closes out the pipeline in nearly all cases, we can introduce context extension methods to not only allow simpler (more intuative) writing of JSON etc but provide the short circuit (Task.FromResult Some ctx) such that handler could instead be written
```fsharp
let handleLunchFilter  _  ctx : HttpHandler =
    let filter = ctx.BindQueryString<LunchFilter>()
    let lunchSpots = LunchAccess.getLunches filter
    ctx.WriteJson(lunchSpots)                  // WriteJson : obj -> Task<HttpContext option>
```

It would not be too much less characters but 

- users get AutoComplete Intellesense for `ctx.WriteXXX()` 
- not having to deal with the less obvious appending of `next ctx` to apply to `HttpHandlers` 
- the user will get an error if they try to write after `.WriteJson` as it is already returning

The exiting HttpHandlers would still remain as they are useful for composed pipeline use vs nested use as displayed above.

I am happy to do PR for this provided everyone happy that it is not providing too much choice / polluting the HttpContext methods.

**Comment by dustinmoris** [comment]
Great suggestion, a PR for this would be nice, thank you!

**Comment by AlbertoDePena** [comment]
@gerardtoconnor 

Any update on the PR? =)

**Comment by gerardtoconnor** [comment]
Apologies, manic week, will put PR in tomorrow

**Comment by AlbertoDePena** [comment]
No worries, thanks =)

**Comment by gerardtoconnor** [comment]
Merged in PR #119
